### PR TITLE
Set PERCY_TARGET_COMMIT to determine which Percy build to use as baseline for PRs

### LINKED ISSUE #11385: Pick the correct baseline for Percy builds on release branches
While running visual tests on a release branch, we currently pick the most recent approved build on `master` as the baseline for diffs. This will result in a false negative if there are approved visual changes in `master` that were merged after the branch point of the release branch.

See https://percy.io/docs/learn/baseline-picking-logic. We want to go from the "branches" behavior to the "pull requests" behavior, and doing so will require the creation of a pull request prior to running tests on a release branch.

**Current behavior:**
branch

**Intended behavior:**
pull

/cc @cramforce @choumx @erwinmombay

### PR DESCRIPTION
**Pull request by danielrozenberg**
This PR sets the PERCY_TARGET_COMMIT in the `gulp visual-diff` task (for non-`master` builds,) which in turn means that PRs will be compared on Percy with the relevant baseline (i.e., where they were split from `master`) as opposed to the "latest master", which might not be relevant to those PRs.

Fixed #11385

### CODE REVIEW

**Comment by danielrozenberg** [comment]
/to @timhaines, continuing the conversation from your project's issue:

We're setting PERCY_BRANCH to `<username>-<branch>` in builds (except for master builds) to more easily differentiate who created each build when reviewing the list of builds on the Percy UI (since we have so many builds every day.) Based on your comment I understand that I should manually set PERCY_TARGET_COMMIT here. This is what this PR is doing

File: build-system/tasks/visual-diff.js [hunk 1]
```diff
@@ -127,6 +127,22 @@ function setPercyBranch() {
...
+ *
+ * This will let Percy determine which build to use as the baseline for this new
+ * build.
+ *
+ * Does not do anything for --master builds, since master builds are always
+ * built on top of the previous commit (we use the squash and merge method for
+ * pull requests.)
+ */
+function setPercyTargetCommit() {
+  if (!argv.master) {
```
Comment (rsimha) [inline build-system/tasks/visual-diff.js]: I take it this will work for local dev too?

**Comment by rsimha** [comment]
Consider addling a logging statement (maybe [here](https://github.com/ampproject/amphtml/blob/8cadedb4cf705be1809a49ed9b793cb26bc4bfc5/build-system/tasks/visual-diff.js#L324)) for the base commit so diffs are easier to understand.

**Comment by timhaines** [comment]
@danielrozenberg After a quick (rather than thorough) look, this looks good to me, with a couple of caveats.

1) If your PR is from a sub-feature-branch into a feature-branch (rather than master), your git merge-base command is going to return an incorrect result, as you have `master` hard coded there.

2) The git merge-base command relies on having a local copy of the master branch.. Depending on how you're retrieving the git repo from GitHub, this may or may not be true.  If you ran `git fetch origin/master` first, and then ran the git merge-base command against origin/master, I think that would work reliably.  (What you have currently may not be problematic - I didn't look in depth)

File: build-system/tasks/visual-diff.js (hunk 1 above)
Comment (danielrozenberg) [reply build-system/tasks/visual-diff.js]: Yep, I don't see why it would be any different

**Comment by danielrozenberg** [comment]
@timhaines thanks for the comments! We recommend a very specific process to contributors, which should give them a local `master` branch in all cases. We also don't really support sub-feature branch development, and as you said, "worst case" they'll be baseline'd against the latest master. I'll keep that in mind if we ever change our MO :)

@rsimha I'm getting a weird result from the `git merge-base` command on this very PR: https://travis-ci.org/ampproject/amphtml/jobs/433240171#L622
This SHA doesn't exist in our repo. It *should* return `5b9874396a8cc44cfb1a344cd38f143cf01a680b`, but it's returning `154e1ffdf95e647a90667253f5569048ba9da7f1` instead. Do you have any idea what's going on?

**Comment by timhaines** [comment]
@danielrozenberg @rsimha Ahh, it's returning `154e1ffdf95e647a90667253f5569048ba9da7f1` because Travis's PR builds run CI off GitHub's temporary merge commits -- which are a merge of master into the branch the PR is for.

**Comment by timhaines** [comment]
@danielrozenberg @rsimha unless you're changing the commit you're running CI on, because CI is being run for the merge commit (and has master changes merged into it), `154e1ffdf95e647a90667253f5569048ba9da7f1` probably is the right commit to pick for the base build..

**Comment by rsimha** [comment]
Hi @timhaines, hope all's well with you :)

Yeah, we already use this function to determine other things like files changed, tests to run, etc. during Travis builds. But I realise there may not be a Percy run associated with that commit if it's not an actual commit on master. I'll look at this tomorrow to see if I can spot the problem / solution.

**Comment by danielrozenberg** [comment]
PTAL, this should work now (I'll double check that it caught the correct commit in the logs before merging)

File: build-system/tasks/visual-diff.js [hunk 2]
```diff
@@ -133,13 +133,13 @@ function setPercyBranch() {
...
- * pull requests.)
+ * Only does something on Travis, and for non-master branches, since master
+ * builds are always built on top of the previous commit (we use the squash and
+ * merge method for pull requests.)
  */
 function setPercyTargetCommit() {
-  if (!argv.master) {
-    process.env['PERCY_TARGET_COMMIT'] = gitBranchPoint();
+  if (process.env.TRAVIS && !argv.master) {
+    process.env['PERCY_TARGET_COMMIT'] = gitBranchPoint(true);
```
Comment (rsimha) [inline build-system/tasks/visual-diff.js]: nit: Make this
```
gitBranchPoint(/* fromMerge */ true)
```

Comment (rsimha) [reply build-system/tasks/visual-diff.js]: Seems like this is obsolete.

**Pull request merged**